runtime/compiler/trj9/runtime/J9ValueProfiler.hpp

```diff
@@ -650,8 +675,9 @@ TR_EmbeddedHashTable<T, bits>::resetLowFreqKeys()
 template <typename T, size_t bits> void
 TR_EmbeddedHashTable<T, bits>::reset()
    {
-   // Clear keys first, checks will only match on zero
-   memset(&_keys, 0, sizeof(T) * _realSize);
+   // Clear keys first, set first entry to -1
+   for (size_t i = 0; i < _realSize; ++i)
+      _keys[i] = getDefault(i);
```

Comment:
For POWER or other weak memory models don't we need to have some kind of fence after this to force the write to be visible to all threads? Otherwise you wouldn't be sure that a given default write had taken place no?

Reply:
Yeah, its not a significant problem as frequencies will be cleared soon after.